### DISCUSSION TITLE
AAN FreeSurfer script

### DISCUSSION THREAD

**Discussion by VictorVidalCuevas** [neurodesk/neurodesk.github.io-hugo#454] [Q&A] [answered]
Dear Neurodesk Team,

I would like to express my sincere gratitude for the outstanding work you have done with Neurodesk. 
I am reaching out with a question regarding the customization of FreeSurfer within Neurodesk. Recently, a new subcortical segmentation (AAN) has been released, and I would like to incorporate a script and related files into the FreeSurfer directory structure. Specifically, I am hoping to add a script to the `bin` directory and other required files to relevant FreeSurfer folders.

Could you please advise on the best way to implement these customizations in Neurodesk?

Thank you again for your continued support and dedication.

Best regards,  
Victor Vidal

**Comment by stebo85** [comment]
Dear @VictorVidalCuevas, 

Thank you for reaching out and the positive feedback :) 

Yes, what you are proposing is quite easy to do: Just send a pull request against this file which adds the things you like and the rest is automatic: https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/freesurfer/build.sh

From your description it sounds like you should be able to copy the workbench lines in our recipe and add the new tool identically :) 

Let us know if you need any more help or hints 

Thank you
Steffen

**Reply by stebo85** [reply]
Dear @VictorVidalCuevas - any updates from your side?

**Comment by VictorVidalCuevas** [comment]
Sorry for the late response, I haven't tested it due other compromises...but as soon as I tested it i'll let you know!
Thanks for your help Steffen!

Best,

**Comment by VictorVidalCuevas** [comment]
Dear Steffen and the NeuroDesk team,

I hope you're doing well!

I’m working on modifying the FreeSurfer build recipe to integrate the AANsegment tool for brain segmentation, as described in the [AANsegment GitHub page](https://github.com/freesurfer/AANsegment). Along with the installation script for AANsegment, I’ve found that there are additional files that need to be included for proper functionality.I would greatly appreciate your guidance on the best way to add both the script and the necessary files to the recipe. Specifically, I need help with the following:

Additional Files: AANsegment requires certain files that are part of the tool’s installation. Could you advise on where these files should be placed within the FreeSurfer recipe? Should they be copied to a specific directory, or is there a standard directory structure I should follow?

Environment Variables: Are there any specific environment variables that need to be set for AANsegment to function properly within the FreeSurfer setup?

Script Integration: Should I integrate the AANsegment installation script directly into the build.sh file, or is it better to create a separate script for this?

This is the first time I’m working on modifying a recipe like this, so I apologize for any mistakes in advance, and for any basic question.

Thank you for your help!
Best regards,
Victor Vidal

**Reply by stebo85** [reply]
Dear Victor,

The link to the AANsegment github page didn't work for me.

But looking at this here:
https://github.com/freesurfer/freesurfer/tree/dev/AANsegment

I would put the files /opt

I don't think it needs any specific Environment Variables

Yes, you need to integrate the steps into the build.sh file, ideally after this line:
https://github.com/NeuroDesk/neurocontainers/blob/08a528be5d2cac2db58b00d7c67eb76cb1c30505/recipes/freesurfer/build.sh#L52

**Comment by VictorVidalCuevas** [comment]
Dear @stebo85 

Thanks for your help! I’ve completed the AANsegment integration and pushed the changes to GitHub. The pull request has been created, and all necessary scripts and files are now included.
Let me know if you need anything else!
Best,
Victor Vidal

**Reply by stebo85** [reply]
Dear @VictorVidalCuevas - thank you for sending through the pull request. The changes made don't work unfortunately: It's not possible to copy files from your computer into the container, so everything has to come from a web resource. Also AANsegment needs the matlab runtime environment which needs to be installed separately. I am currently trying if the 8.0.0 beta release would work as a starting point: https://surfer.nmr.mgh.harvard.edu/fswiki/rel7downloads

**Reply by VictorVidalCuevas** [reply]
Dear @stebo85

Oh I see, thanks for let me know!
Thanks for your response and time, I appreciate it!

Best,

**Reply by stebo85** [reply]
AANsegment doesn't work yet in 8.0.0 - I opened an issues on the freesurfer github and hope there is a fix soon.

**Reply by stebo85** [reply]
Dear @VictorVidalCuevas - I finally got it to work. The new container is building here: https://github.com/NeuroDesk/neurocontainers/actions/runs/12944747792

**Reply by VictorVidalCuevas** [reply]
> Dear @VictorVidalCuevas - I finally got it to work. The new container is building here: https://github.com/NeuroDesk/neurocontainers/actions/runs/12944747792

Thank you so much for your time and effort @stebo85, I'll test it as soon as possible. 
Cheers

**Reply by stebo85** [reply, marked as answer]
Dear @VictorVidalCuevas - the new freesurfer 8.0.0 container is now released and you can get it by clicking the update button in the Neurodesk menu. segmentNuclei appears to be working ok. Can you test and confirm?
<img width="737" alt="image" src="https://github.com/user-attachments/assets/7c940ba0-3396-4cd9-a976-cb2d99590e4a" />
